adobe/copy_on_write.hpp

```diff
@@ -222,176 +217,36 @@ class copy_on_write {
     \return Boolean; <code>true</code> if the underlying object instance is
     shared by both objects.
     */
-    bool identity(const copy_on_write& x) const { return object_m == x.object_m; }
+    bool identity(const copy_on_write& x) const noexcept { return _self == x._self; }
 
-    friend inline void swap(copy_on_write& x, copy_on_write& y) {
-        std::swap(x.object_m, y.object_m);
+    friend inline void swap(copy_on_write& x, copy_on_write& y) noexcept {
+        std::swap(x._self, y._self);
     }
 
-    friend inline bool operator<(const copy_on_write& x, const copy_on_write& y) {
-        return y.object_m && (!x.object_m || (!x.identity(y) && *x < *y));
+    friend inline bool operator<(const copy_on_write& x, const copy_on_write& y) noexcept {
+        return !x.identity(y) && (*x < *y);
```

Comment:
This will assert on the call to identity() if either is empty. I tried to keep the asserts to only where _self is dereferenced (and missed a place or two).